ARCdigi/src/ARCdigitizer.cpp

```diff
@@ -0,0 +1,87 @@
+#include "ARCdigitizer.h"
+
+// EDM4HEP
+#include "edm4hep/Vector3d.h"
+
+// DD4HEP
+#include "DDRec/CellIDPositionConverter.h"
+
+// ROOT
+#include "TRandom.h"
+
+// STL
+#include <unordered_map>
+#include <utility>
+
+DECLARE_COMPONENT(ARCdigitizer)
+
+ARCdigitizer::ARCdigitizer(const std::string& aName, ISvcLocator* aSvcLoc) : GaudiAlgorithm(aName, aSvcLoc) {
+  declareProperty("inputSimHits", m_input_sim_hits, "Input sim tracker hit collection name");
+  declareProperty("outputDigiHits", m_output_digi_hits, "Output digitized tracker hit collection name");
+  declareProperty("detectorCompact", m_det_compact, "Path to detector compact");
+  declareProperty("flatSiPMEfficiency", m_flat_SiPM_effi, "Flat value for SiPM efficiency (<0 := disabled)");
+  declareProperty("applySiPMEffiToDigiHits", m_apply_SiPM_effi_to_digi,
+                  "Apply the SiPM efficiency to digitized hits instead of simulated hits");
+}
+
+ARCdigitizer::~ARCdigitizer() {}
+
+StatusCode ARCdigitizer::initialize() {
+  m_detector = dd4hep::Detector::make_unique(this->name() + "_detector");
+  if (m_det_compact.value().empty()) {
+    error() << "Detector compact must be provided!" << endmsg;
+    return StatusCode::FAILURE;
+  }
+  if (m_flat_SiPM_effi > 1.0) {
+    error() << "Flat SiPM efficiency cannot exceed 1!" << endmsg;
+    return StatusCode::FAILURE;
+  }
+  m_detector->fromCompact(m_det_compact);
+  return StatusCode::SUCCESS;
+}
+
+StatusCode ARCdigitizer::execute() {
+  // Get the input collection with Geant4 hits
+  const edm4hep::SimTrackerHitCollection* input_sim_hits = m_input_sim_hits.get();
+  verbose() << "Input Sim Hit collection size: " << input_sim_hits->size() << endmsg;
+
+  // Dictionary to keep track of cell IDs and (summed) deposited energies / (earliest) arrival times
+  std::unordered_map<uint64_t, std::pair<float, float>> merged_digi_hits;
+
+  // Digitize the sim hits
+  TRandom rand = TRandom();
```

Comment:
For reproducibility and homogeneity, the random number generation from Gaudi should be used. See e.g. [this example](https://github.com/HEP-FCC/k4RecCalorimeter/blob/4704fb695a1403c33e1053c8096af9feccb0148f/RecCalorimeter/src/components/NoiseCaloCellsFromFileTool.h#L91)

Reply:
Done1